eclair-core/src/test/scala/fr/acinq/eclair/channel/RegisterSpec.scala

```diff
@@ -0,0 +1,30 @@
+package fr.acinq.eclair.channel
+
+import fr.acinq.eclair._
+
+import scala.concurrent.duration._
+import akka.actor.{ActorRef, ActorSystem, Props}
+import akka.testkit.TestProbe
+import fr.acinq.bitcoin.ByteVector32
+import fr.acinq.bitcoin.Crypto.PublicKey
+import org.scalatest.funsuite.AnyFunSuite
+import akka.pattern.ask
+import akka.util.Timeout
+
+import scala.concurrent.Await
+
+class RegisterSpec extends AnyFunSuite {
+
+  case class CustomChannelRestored(channel: ActorRef, channelId: ByteVector32, peer: ActorRef, remoteNodeId: PublicKey) extends AbstractChannelRestored
+
+  test("register processes custom restored events") {
+    implicit val timeout: Timeout = Timeout(3.seconds)
+    implicit val system: ActorSystem = ActorSystem("test")
+
+    val registerRef = system.actorOf(Props(new Register))
+    val customRestoredEvent = CustomChannelRestored(TestProbe().ref, randomBytes32(), TestProbe().ref, randomKey().publicKey)
+    registerRef ! customRestoredEvent
+
+    assert(Await.result(registerRef ? Symbol("channels"), 3.seconds).asInstanceOf[Map[ByteVector32, ActorRef]].keys.head == customRestoredEvent.channelId)
```

Comment:
You should use `awaitCond` to avoid blocking with `Await`. This is done in many other tests if you need examples.

And your test should be more thorough: you should wait until the register returns a map which contains `customRestoredEvent.channelId` as key, AND that the value is the right actor ref.